docs/natural-history-inputs.md

```diff
@@ -0,0 +1,95 @@
+are associated with longer times to symptom improvement -- or there may not be. By having the user specify
+a CSV file which contains all of these parameter sets, not only does this mean that the user determines
+the distribution of each of these parameters outside of Ixa, but the user is also able to impose any
+meaningful correlations between parameters.
+
+## Data input format
+
+The input CSV includes samples of the natural history parameters. For now, we are interested in a single
+natural history parameter: the generation interval CDF over time. Data must be input in a long format, as
+such:
+
+| id | time | gi_cdf |
+| --- | --- | --- |
+| 0 | 0.0 | 0.0 |
+| 0 | 1.0 | 0.5 |
+| 0 | 2.0 | 1.0 |
+| 1 | 0.0 | 0.0 |
+| 1 | 1.0 | 0.3 |
+| ... | ... | ... |
+
+The `id` is an identifier that marks a distinct sample from the natural history parameters at a given `time`
+since the person was first infected. The `gi_cdf` describes the fraction of infectiousness that has occured
+in this parameter set at time t. Additional columns like `viral_load` and other time-varying quantities
+can easily be added in this input schema.
+
+In the future, we may also add variables like `symptom_onset_time` and `symptom_improvement_time`. These
+variables are constant in time, so the same value will be present for all rows of a given `id` value.
+If a person is asymptomatic, their `symptom_onset_time` and `symptom_improvement_time` should be NA.
+This choice allows for one input CSV that includes both symptomatics and asymptomatics, and the user
+can specify different GIs for symptomatics and asymptomatics with great flexibility.
+
+Finally, this input schema naturally lends itself to adequately modeling per-person variability in
+the natural history parameters. The user can specify as many different natural history parameter
+sets (potentially where each parameter set has completely different values for all parameters or
+where each parameter set has, say, the same GI but different symptom onset times) as needed to properly
+reproduce the underlying parameter distribution, and the model will sample from these values accordingly.
+
+Note that this input structure also means that each parameter set can have parameter samples at
+different times. This could be useful if synthesizing various clinical datasets that each record the
+viral load at different times and a user wants to include the raw data in the model without doing
+any pre-processing, though it is worth noting that the module is set up to use cubic spline
+interpolation for intermediary values (more on this below.)
+
+## Implementation
+
+People are assigned an `id` when they are infected. This `id` refers to the natural history parameter
+set that will define their infection (for now, just the GI, but this will be expanded to include
+symptom onset and improvement times). `id`s are assigned uniformly and randomly, which allows for
```

Comment:
Are there any covariates for the sampling? For instance, I thought different symptoms were correlated with different type of curves, how are we handling that?